### PR TITLE
Refactor registrar

### PR DESCRIPTION
**Pull request by ruflin**
Currently the state is in the crawler but should be part of the registrar. This will be a small refactoring.

### CODE REVIEW

**Comment by monicasarbu** [comment]
LGTM

File: crawler/registrar.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+	r := &Registrar{
+		registryFile: registryFile,
+	}
+	r.Init()
+
+	return r
+}
+
+func (r *Registrar) Init() {
```
Comment (urso) [inline crawler/registrar.go]: why not Init(registryFile string) ?

Then New becomes:

    r := Registrar{}
    r.Init(registryFile)
    return &r

Comment (urso) [inline crawler/registrar.go]: Init is exported. => ensure Init on zero value has same result as using New?

Comment (monicasarbu) [inline crawler/registrar.go]: I agree with @urso.

Comment (ruflin) [inline crawler/registrar.go]: Also an option. I was thinking that init just does the default values, and NewRegistrar needs to be used to spcify a specific files. Probably we should have a general convention across the hole project for such things. Something like: X method is expected to do Y, Z is expected to do Q.

Comment (urso) [inline crawler/registrar.go]: +1

Comment (andrewkroh) [inline crawler/registrar.go]: Agree, a convention would be great.

**Pull request merged**